### PR TITLE
docs(mesh): add document for production usage helm values

### PR DESCRIPTION
**Pull request by jijiechen**
### Description

 

fixes https://github.com/Kong/docs.konghq.com/issues/7507

### Testing instructions

Preview link: 

https://deploy-preview-7508--kongdocs.netlify.app/mesh/dev/production/cp-deployment/production-usage-values/
https://deploy-preview-7508--kongdocs.netlify.app/mesh/latest/production/cp-deployment/production-usage-values/

### Checklist 

- [x] Review label added 
- [x] [Conditional version tags](https://docs.konghq.com/contributing/conditional-rendering/#conditionally-render-content-by-version) added, if applicable.

### CODE REVIEW

File: app/_src/mesh/production/cp-deployment/production-usage-values.md [hunk 1]
```diff
@@ -0,0 +1,755 @@
...
+
+The values in this page may reference resources that need to be created in advance when certain features are enabled, please read the value files carefully and prepare these resources according to notes near the `(action)` keywords. If you decide to disable the feature that require a resource to be pre-existing, please remove/change those fields according to the full [reference of helm configuration](/mesh/{{ page.release }}/reference/kuma-cp/#helm-valuesyaml).
+
+Please choose one `values.yaml` file according to your installation scenario:
+
+<!-- v2.7.x and below does not support controlPlane.defaults.adminRoleGroups -->
+{% if_version gte:2.8.x %}
+{% tabs values-yaml useUrlFragment=true %}
+{% tab values-yaml single zone control plane %}
+```yaml
```
Comment (lahabana) [inline app/_src/mesh/production/cp-deployment/production-usage-values.md:32]: I'm a little concerned by the supportability of this. Should we instead land them in the config that gets synced and use embed? Like we do for kuma-cp.yaml?

Comment (jijiechen) [reply app/_src/mesh/production/cp-deployment/production-usage-values.md:32]: Yeah, it's a better idea to generate them from the product source code. I'm thinking how to extract this information and embed them here.
Let me see if there are existing similar mechanisms.

Comment (jijiechen) [reply app/_src/mesh/production/cp-deployment/production-usage-values.md]: I've changed the content and reference from embeded files.
So this PR will only contain the text and the code block wrappers.

Comment (jijiechen) [reply app/_src/mesh/production/cp-deployment/production-usage-values.md]: I'll push changes to kong-mesh to make sure those files are synced to this repo.

**Comment by jijiechen** [comment]
I'm struggling analyzing why the smoke-test fails. Could someone help?

**Comment by jijiechen** [comment]
I'll try to update the code after this PR is merged so that the embeded files are correctly included: https://github.com/Kong/kong-mesh/pull/6131

**Comment by jijiechen** [comment]
I believe the CI task "Run Tests / smoke-test" can pass for PRs from forks. Can you confirm this? I think it should be fixed for the public who can't file PRs from branches within this repo. @fabianrbz 

The error must be caused by an empty value for the variable "VITE_PORTAL_API_URL". 
https://github.com/Kong/docs.konghq.com/actions/runs/9545789640/job/26307313676?pr=7508#step:12:7

**Comment by cloudjumpercat** [comment]
@jijiechen You're correct, the smoke test fails on forked PRs. 

@fabianrbz Is that something we could fix? Or maybe there's a label we could add to auto pass the smoke tests on forks?

**Review by cloudjumpercat — CHANGES_REQUESTED** [review]
@jijiechen Thanks for creating this page, this will be really helpful! 

Couple of questions:

- Does this page only apply to Kong Mesh? I wasn't sure if this should also be in the Kuma docs.
- I couldn't push to your fork, so I left my suggested edits as a comment, sorry about the bad formatting.
- Does this apply to versions earlier than 2.8? Just wondering if we could add this to the other navs as well.

**Comment by jijiechen** [comment]
Sorry for the late reply
1. It Kong Mesh only, because we decided this to be an paid advice
2. Thanks for the suggestions, I'll work on it once I get some time
3. It's only for 2.8 at this time, and we'll figure out how can we add it for previous versions

File: app/_data/docs_nav_mesh_2.8.x.yml [hunk 2]
```diff
@@ -51,6 +51,11 @@ inherit:
       index: 0
       text: Kong Mesh license
       url: /production/cp-deployment/license
+    - path: [ Kong Mesh in Production, Control plane deployment ]
```
Comment (lahabana) [inline app/_data/docs_nav_mesh_2.8.x.yml:54]: Should be added in the 2.9 sidebar as well now

**Comment by jijiechen** [comment]
I can't preview the site on my local env for some reason and netlify also refuse to verify my account to proceed the preview deployoment.
So I can't make any preview at this time. 

image

**Comment by jijiechen** [comment]
I opened a [new PR](https://github.com/Kong/docs.konghq.com/pull/7640) from this repo to pass the CI.
So  closing this one.

**Pull request closed without merging**